iceoryx_hoofs/include/iceoryx_hoofs/cxx/optional.hpp

```diff
@@ -80,7 +83,7 @@ class optional : public FunctionalInterface<optional<T>, T, void>
     ///         optional via .value() or the arrow operator the application
     ///         terminates.
     // NOLINTNEXTLINE(hicpp-explicit-conversions) for justification see doxygen
-    optional(const nullopt_t& noValue) noexcept;
+    optional(const nullopt_t noVal) noexcept;
```

Comment:
I think the clean C++ way is to remove the variable name. With this the compiler knows that this is not used.
Removing the reference is the wrong approach.

Reply:
Since the input argument is unused hence the reference was removed, but the suggestion looks more apt here. thanks